### PR TITLE
Changes Flashbang and Sonic Powder to use adjustEarDamage(), Adds AI Check

### LINKED ISSUE #4483: AIs are made permanently deaf by flashbangs
Also, an AI deafened by a flashbang can't use binary, which doesn't make much sense.

### PR DESCRIPTION
**Pull request by Twinmold93**
This changes flashbangs and sonic powder from using direct damage adjustments to the adjustEarDamage() system. Adds a check that makes it impossible to deafen AIs, but not borgs. Multiple flashbangs/sonic powder reactions now stacks damage. Technically rebalances these.

Original: Fixes AI's from being Able to be Deafened
Fixes the ability to make AI's go deaf from flashbangs, sonic powders, or anything else that can cause ear damage by making it just revert back to null, the original value.

Fixes #4483

:cl: Twinmold
Fix: AI's can no longer be made deaf.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/silicon/ai/life.dm [hunk 1]
```diff
@@ -158,6 +158,8 @@
 			health = 100 - getOxyLoss() - getToxLoss() - getFireLoss() - getBruteLoss()
 		diag_hud_set_status()
 		diag_hud_set_health()
+		ear_damage = null
```
Comment (tigercat2000) [inline code/modules/mob/living/silicon/ai/life.dm]: both of these should be `0` not `null`
`null++` would break

Comment (Twinmold93) [inline code/modules/mob/living/silicon/ai/life.dm]: @tigercat2000 Strangely enough, as the code currently stands, and you can check in AI, when the value is null, the game automatically makes it a number when they get any ear damage. As the game sits, all AI start with ear damage of null, and nor do they break. when they get a value added.

Comment (tigercat2000) [inline code/modules/mob/living/silicon/ai/life.dm]: AI code. Why am I suprised?
Still, though, should be 0, it's _supposed_ to be a number..

**Comment by Fox-McCloud** [comment]
It would be better to have a proc for adjusting ear damage and deafness, then utilizing that proc to change ear damage instead of manipulating the vars directly.

Then, you can just have that proc `return` for silicons.

This is better than handling it in the already horrendous `Life` loop.

File: code/modules/mob/living/silicon/ai/life.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/modules/mob/living/silicon/ai/life.dm]: @tigercat2000 ...a lot off vars...like weakened, dizzines, etc...are all null by default, not 0.

I have no idea why.

**Comment by Twinmold93** [comment]
Side note: That would require us changing everything that utilizes a direct affect (which everything currently does). That means the change take a bit longer.

**Comment by Ar3nn** [comment]
@Fox-McCloud As above, copy-pasta code. Ew. As far as I know, you'd have to define that proc in everything that deals with ear damage separately, then run the proc in everything.

**Comment by Twinmold93** [comment]
It already exists as a non-proc elsewhere, under code/modules/mob/living/living.dm but doesn't affect AIs at all, since it doesn't fall under that for silicons for some strange reason. That means, unless it's in AI itself, it will not function correctly.

**Comment by Fox-McCloud** [comment]
Heh...so we have it, it's just not plugged in or overridden for silicons.

**Comment by Twinmold93** [comment]
Yeah. That's the problem. It just doesn't process for silicons in the first place, for some reason. I could probably add to life for AI setEarDamage(0, 0) to make it a line less, that way borgs can still go deaf. Less we want to make it so no silicons can go deaf at all.

**Comment by Fox-McCloud** [comment]
=V

No. `Life` is already bad enough, as is, we don't need stuff liked this added to it when there's better solutions. I can just convert us to an ear damage proc in another PR, if you like--just let me know and I'll take care of it and close this (or you can).

**Comment by Twinmold93** [comment]
What would be a better solution? The proc exists in
https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/mob/mob.dm
itself, the connection in living, and yet silicon never gets referenced for it at all.

**Comment by Fox-McCloud** [comment]
ideally everything should use that proc to adjust ear damage and such then be overridden when we don't want a particular mob type to be impacted by ear damage at all.

Like I said, I can make that PR if you like.

**Comment by Twinmold93** [comment]
AKA - Find every source that uses its own form of damaging ears, and switch all of them to adjustEarDamage(_,_). Wonder where explosions hold that, since reagents and flashbangs are the only other ones that need the change (that I know of). Then again, I've never heard an AI go deaf to an explosion.

**Comment by Fox-McCloud** [comment]
Oh, nevermind, the base proc of it does nothing.

I'll just fix this in my cleanup PR.

**Comment by Twinmold93** [comment]
I'm doing a test with changing flashbangs and sonic powder to using adjustEarDamage rather than their current standard of forced adjustment as well as with the check for silicon in the proc itself.

**Comment by Fox-McCloud** [comment]
Alrighty, closing this, as I have it handled in my cleanup PR, since it doesn't really impact balance or anything.

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -282,11 +282,13 @@
 
 //damage/heal the mob ears and adjust the deaf amount
 /mob/living/adjustEarDamage(damage, deaf)
+	if(istype(src, /mob/living/silicon/ai)) return
```
Comment (Fox-McCloud) [inline code/modules/mob/living/living.dm]: *glares*

NO. Never do this. You override a proc like this

```
/mob/living/silicon/adjustEarDamage()
     return
```

**Pull request closed without merging**